src/Endatix.Infrastructure/Integrations/Slack/SlackClient.cs

```diff
@@ -0,0 +1,68 @@
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Endatix.Core.Entities;
+using Endatix.Core.Events;
+using Endatix.Core.Infrastructure.Domain;
+using Endatix.Core.Integrations.Slack;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Endatix.Infrastructure.Integrations.Slack;
+/// <summary>
+/// A client for communications with the Slack API.
+/// </summary>
+public class SlackClient : INotificationHandler<SubmissionCompletedEvent>, ISlackClient
+{
+    private readonly ILogger<SlackClient> _logger;
+    private readonly IRepository<Form> _formRepository;
+    private readonly SlackSettings _slackSettings;
+    public SlackClient(ILogger<SlackClient> logger, IRepository<Form> formRepository, IOptions<SlackSettings> options) {
+        _formRepository = formRepository;
+        _logger = logger;
+        _slackSettings = options.Value;
+    }
+
+    public async Task Handle(SubmissionCompletedEvent notification, CancellationToken cancellationToken) {
+        if(notification.Submission.IsComplete && _slackSettings.Active){
```

Comment:
Not something necessarily needed but a suggestion. Having a not completed submission on SubmissionCompletedEvent is not an expected situation. Logging a warning in such case would bring attention if it happens.

Reply:
Done